web/src/fetch_status.ts

```diff
@@ -70,7 +70,7 @@ export class FetchStatus {
     }
 
     finish_newer_batch(
-        messages: RawMessage[],
+        messages: Message[],
```

Comment:
Empirically this is wrong.

Like Tim said, a `RawMessage` is what we get directly from the server, and it’s mutated into a `Message` by `message_helper.process_new_message`. (That mutation concept is going to need to change somewhat because transmuting one type into another doesn’t make sense from the perspective of TypeScript or the perspective of basic sanity, but it’s what we have.)

`finish_newer_batch` is called directly from `message_fetch.load_messages` → `get_messages_success`, which uses the unprocessed data from `GET /json/messages`. That happens before the call to `get_messages_success` → `process_result` → `message_helper.process_new_message`. So at this time, we’re still dealing with `RawMessage`.

Reply:
Changed `max_id_for_messages` instead to accept `RawMessage`